crates/tako/src/internal/server/reactor.rs

```diff
@@ -133,10 +132,34 @@ pub(crate) fn on_remove_worker(
         worker_id,
         running_tasks
     );
-
     let _ = core.remove_worker(worker_id);
+
     comm.broadcast_worker_message(&ToWorkerMessage::LostWorker(worker_id));
+
+    // IMPORTANT: We have to announce error BEFORE we announce lost worker (+ running tasks)
```

Comment:
Maybe we should fix HQ instead? As long as we pretend that Tako doesn't know anything about HQ :)

Reply:
Ok. As long as the messages cannot go out of order, then I guess it's fine for this usecase.